apps/settings/js/modules/sim_security.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * This is a simple model that can help us mimic the status change event
+ * when enabling / disabling simcard.
+ *
+ * @module SimSecurity
+ */
+define(function(require) {
+  'use strict';
+
+  var Module = require('modules/base/module');
+  var SettingsUtils = require('modules/settings_utils');
+  var EventEmitter = require('modules/base/event_emitter');
+
+  var EVENT_NAME = 'simpin';
+  var APIS = [
+    'unlockCardLock',
+    'setCardLock',
+    'updateContact',
+    'getCardLock',
+    'getCardLockRetryCount'
+  ];
+
+  var EVENT_CHANGE_APIS = [
+    'unlockCardLock',
+    'setCardLock'
+  ];
+
+  var SimSecurity = Module.create(function() {
+    this.super(EventEmitter).call(this, [
+      EVENT_NAME + '-pin-enabled',
```

Comment:
nit: The name itself seems pretty self-explanatory and the prefix may not be required.